### DISCUSSION TITLE
Thermostat set points

### DISCUSSION THREAD

**Discussion by quadnegative** [zwave-js/zwave-js-ui#2368] [Q&A]
I have a CT100 thermostat. I used to use Vera for zwave with HA. I used to be able to see my temperature set points regardless of the mode. However, since I switched to zwavejs2mqtt, the set points and HA entities are nulled out when the operation mode is set to off.
![image](https://user-images.githubusercontent.com/13278661/163067885-34d04ea6-721e-41ce-a6f7-995579cea071.png)
Yet, the set points are still in zwavejs
![image](https://user-images.githubusercontent.com/13278661/163068059-5cca694c-afc8-423d-947d-980895dda93e.png)
I used to be able to do automations based on the entity values, but now that they are being nulled, everything fails.
Am I missing something?

**Comment by robertsLando** [comment]
@quadnegative are you using mqtt discovery or zwavejs server with hass integration?

**Reply by quadnegative** [reply]
As far as I know, zwavejs server with hass integration. I installed the zwavejs2mqtt addon in HA and it installed both zwavejs and zwavejs2mqtt.

**Comment by kpine** [comment]
The target temperature values are based on the thermostat's current mode. If the thermostat is off, then those values are not available, since there is currently no target temperature. That is how many of the HA integrations are implemented.

**Reply by quadnegative** [reply]
That sounds odd.  I used to be able to set the mode based on my presence or math based on the target temps. Now I have to hard set temps in code when my presence changes.
This seems like not being able to get the sun elevation because it is cloudy or night. All entities should be available regardless of mode.

**Reply by kpine** [reply]
The target temperatures can change depending on the mode being used. If you're using an away mode for example, different setpoints.

**Reply by kpine** [reply]
You can setup MQTT and access the setpoints directly using MQTT template entities. The integration does not provide an easier way to do that.
